test/osutils.jl

```diff
@@ -0,0 +1,8 @@
+@test (@static true ? 1 : 2) == 1
```

Comment:
i'm easy.  by putting it in test/osutils.jl i was just trying to mimic the pattern that there is generally a 1:1 correspondence between base/*.jl and test/*.jl.  will make the change as soon as the current CI finishes so i know if there are any other problems.